### PR TITLE
datefudge: work correctly even if GNU date is not in PATH

### PR DESCRIPTION
**Pull request by KAction**
Examples in manual assumes advanced features from date(1) like "last
Friday", which only available in GNU coreutils version of date(1)
utility. Without this patch, most examples from datefudge(1) manual will
fail in busybox environment, which is confusing.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by KAction** [comment]
@ma27 @leenaars

**Comment by dasJ** [comment]
Why not go for a patchless approach and use `wrapProgram`?

**Comment by SuperSandro2000** [comment]
I agree with @dasJ, please wrap the program.

**Comment by KAction** [comment]
@SuperSandro2000 @dasJ Applied changes.

File: pkgs/tools/system/datefudge/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchgit, fetchpatch }:
+{ stdenv, lib, fetchgit, fetchpatch, makeWrapper, coreutils ? null }:
```
Comment (SuperSandro2000) [inline pkgs/tools/system/datefudge/default.nix:1]: I would prefer if we just always pass in coreutils if this is possible.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 94045` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>datefudge</li>
  </ul>
</details>

File: pkgs/tools/system/datefudge/default.nix (hunk 1 above)
Comment (KAction) [reply pkgs/tools/system/datefudge/default.nix:1]: On build system -- yes, coreutils is part of stdenv. On host system -- not necessary.

File: pkgs/tools/system/datefudge/default.nix [hunk 2]
```diff
@@ -20,7 +21,11 @@ stdenv.mkDerivation rec {
 
   installFlags = [ "DESTDIR=$(out)" ];
 
-  postInstall = "chmod +x $out/lib/datefudge/datefudge.so";
+  postInstall = ''
+    chmod +x $out/lib/datefudge/datefudge.so
+  '' + lib.optionalString (coreutils != null) ''
+    wrapProgram $out/bin/datefudge --prefix PATH : ${coreutils}/bin
```
Comment (SuperSandro2000) [inline pkgs/tools/system/datefudge/default.nix:27]:
```suggestion
    wrapProgram $out/bin/datefudge --prefix PATH : ${coreutils}/bin
```

File: pkgs/tools/system/datefudge/default.nix (hunk 1 above)
Comment (SuperSandro2000) [inline pkgs/tools/system/datefudge/default.nix:1]:
```suggestion
{ stdenv, lib, fetchgit, fetchpatch, makeWrapper, coreutils }:
```

File: pkgs/tools/system/datefudge/default.nix [hunk 3]
```diff
@@ -9,6 +9,7 @@ stdenv.mkDerivation rec {
     rev = "debian/${version}";
     sha256 = "1nh433yx4y4djp0bs6aawqbwk7miq7fsbs9wpjlyh2k9dvil2lrm";
   };
+  nativeBuildInputs = lib.optional (coreutils != null) makeWrapper;
```
Comment (SuperSandro2000) [inline pkgs/tools/system/datefudge/default.nix:12]:
```suggestion
  nativeBuildInputs = [ coreutils makeWrapper ];
```

**Comment by KAction** [comment]
Any chance to merge it as I wrote it? I still maintain that avoiding unnecessary dependency worth two conditional statements.

File: pkgs/tools/system/datefudge/default.nix [hunk 4]
```diff
@@ -9,6 +9,9 @@ stdenv.mkDerivation rec {
     rev = "debian/${version}";
     sha256 = "1nh433yx4y4djp0bs6aawqbwk7miq7fsbs9wpjlyh2k9dvil2lrm";
   };
+  nativeBuildInputs = [ makeWrapper ];
+
```
Comment (SuperSandro2000) [inline pkgs/tools/system/datefudge/default.nix:13]:
```suggestion
  };

  nativeBuildInputs = [ makeWrapper ];

```

**Pull request merged**